src/Microsoft.AspNetCore.Rewrite/RewriteMiddleware.cs

```diff
@@ -75,25 +76,13 @@ public Task Invoke(HttpContext context)
 
             foreach (var rule in _options.Rules)
             {
-                rule.ApplyRule(rewriteContext);
-                var currentUrl = new Lazy<string>(() =>
+                var iisRule = rule as IISUrlRewriteRule;
+                if (iisRule != null)
                 {
-                    if ((rule as IISUrlRewriteRule)?.Global == true)
-                    {
-                        var uriBuilder = new UriBuilder(context.Request.Scheme, context.Request.Host.Host);
-                        if (context.Request.Host.Port.HasValue)
-                        {
-                            uriBuilder.Port = context.Request.Host.Port.Value;
-                        }
-                        uriBuilder.Path = context.Request.Path;
-                        if (context.Request.QueryString.HasValue)
-                        {
-                            uriBuilder.Query = context.Request.QueryString.Value;
-                        }
-                        return uriBuilder.Uri.AbsoluteUri;
-                    }
-                    return context.Request.Path + context.Request.QueryString;
-                });
+                    rewriteContext.GlobalRule = iisRule.Global;
```

Comment:
See comment above, I feel like during parsing, we can set if an IIS rule is a global rule or not.